### PR TITLE
fix: Example Body Content Overflow with support for keyboard scroll

### PR DESCRIPTION
**Pull request by bstanton678**
Added tabindex to the scrollable area of this example.

## Summary

Fixes: #2222  

Keyboard accessibility: the scrollable area in the Body Content Overflow example of a modal could not be scrolled up/down using the keyboard only. 

## Release Category
Component Examples

### Release Note
Adding tabIndex 0 to Modal.Body will allow for content to be accessible and scrollable if the content overflows

---

## Checklist

- [ ] MDX documentation adheres to Canvas Kit's [standard MDX template](https://github.com/Workday/canvas-kit/discussions/1131)
- [ ] Label `ready for review` has been added to PR

## For the Reviewer

- [ ] PR title is short and descriptive
- [ ] PR summary describes the change (Fixes/Resolves linked correctly)
- [ ] PR Release Notes describes additional information useful to call out in a release message or removed if not applicable
- [ ] Breaking Changes provides useful information to upgrade to this code or removed if not applicable

## Where Should the Reviewer Start?

1. I would open the modal example, and inspect the scrollable. overflow container for a tabindex attribute. 
2. When the modal is opened, verify keyboard focus is set to the top of the dialog, 
3. Use the Tab key to focus the scrollable area, 
4. Verify the Up / Down arrow keys can be used to scroll all content into view. 

## Areas for Feedback? (optional)

- [ ] Code
- [ ] Documentation
- [ ] Testing
- [ ] Codemods

## Screenshots or GIFs (if applicable)

This should not have much affect on the UI. There will be a visible focus border displayed around the scrollable area.

### CODE REVIEW

File: modules/react/modal/stories/examples/BodyOverflow.tsx [hunk 1]
```diff
@@ -20,7 +20,7 @@ export const BodyOverflow = () => {
         <Modal.Card>
           <Modal.CloseIcon aria-label="Close" />
           <Modal.Heading>MIT License</Modal.Heading>
-          <Modal.Body>
+          <Modal.Body tabIndex="0">
```
Comment (mannycarrera4) [inline modules/react/modal/stories/examples/BodyOverflow.tsx:23]:
```suggestion
          <Modal.Body tabIndex={0}>
```

**Comment by mannycarrera4** [comment]
I'm wondering if this be the default and always add tabIndex 0 to the body instead of in the example?

**Comment by bstanton678** [comment]
> I'm wondering if this be the default and always add tabIndex 0 to the body instead of in the example?

I thought a little bit about that too, but I only want the Modal.Body to be keyboard focusable when the content overflows like in this example. The Basic example is great as-is, and I don't want to focus body content there. 

Maybe the move is to support a new boolean prop for Modal.Body for consumers?

**Comment by mannycarrera4** [comment]
> > I'm wondering if this be the default and always add tabIndex 0 to the body instead of in the example?
> 
> I thought a little bit about that too, but I only want the Modal.Body to be keyboard focusable when the content overflows like in this example. The Basic example is great as-is, and I don't want to focus body content there.
> 
> Maybe the move is to support a new boolean prop for Modal.Body for consumers?

I think this is where a great call out in the documentation would exists saying if you need focus and have scrollable content, use this tabIndex. I agree that the example as is is good for now

**Pull request merged**